### PR TITLE
[READY] Weaving Skill And Silk

### PR DESCRIPTION
**Pull request by SELFHELL**
## About The Pull Request

TLDR: Adds the weaving skill, a bunch of silk clothing, silk dresses/coats, webbed clothing, and a couple other things.

Partially ports https://github.com/Blackstone-SS13/BLACKSTONE/pull/558

Ports https://github.com/Blackstone-SS13/BLACKSTONE/pull/1012

Ports https://github.com/Blackstone-SS13/BLACKSTONE/pull/1062

## Why It's Good For The Game

Allows people to make "exotic" clothing befitting of the upper class.
Also more crafting options and stuff.

100% Gonna need a TM or two

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#define TESTING				//By using the testing("message") proc you can create debug-feedback for people with this
+//#define TESTING				//By using the testing("message") proc you can create debug-feedback for people with this
```
Comment (BadAtThisGame302) [inline code/_compile_options.dm:1]: Is this meant to be here?

Comment (SELFHELL) [reply code/_compile_options.dm:1]: ah shit, no

**Comment by cornercube** [comment]
make the silk half-cloak white instead of black to differentiate it from the regular half cloak

or make the silk half cloak recipe a lot more efficient than the regular recipe, to loosely show that silk is a lot stronger and needs less material

**Comment by SELFHELL** [comment]
right then.. to fix this

**Comment by SELFHELL** [comment]
Conflicts fixed

**Comment by hermaplusplus** [comment]
@maaacha please squash merge this PR once you've reviewed and approved it.

**Comment by SELFHELL** [comment]
conflicts fixed again
